### PR TITLE
feat(platform)!: configurable epoch time (default down to 9.125 days)

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
On Testnet, one of the most important things to test is the events that happen at epoch change. Right now we just don't see it happen very often as it was happening every 18.25 days. It is now configurable. The default config is 9.125 days now (which is what we are planning to now use for Mainnet). There will also now be 40 epochs per era instead of 20.

We will now be able to set the time to 1 hour or so and get a lot of epochs, and be able to test many events far more easily.

## What was done?
Made epoch length configurable. Changed "year" to era. So the length of time transactions are paid for could also be configurable.

## How Has This Been Tested?
Added a strategy tests, modified others.

## Breaking Changes
This will be breaking, as we change the default epoch length

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [X] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [X] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by QuantumExplorer** [comment]
Dashmate test is failing on all PRs. Ignoring.

File: packages/rs-drive-abci/src/execution/platform_events/block_fee_processing/process_block_fees/v0/mod.rs [hunk 1]
```diff
@@ -243,8 +244,9 @@ mod tests {
             let current_epoch = Epoch::new(epoch_index).unwrap();
             let platform_version = PlatformVersion::latest();
 
-            let block_time_ms =
-                genesis_time_ms + epoch_index as u64 * EPOCH_CHANGE_TIME_MS_V0 + block_height;
+            let block_time_ms = genesis_time_ms
+                + epoch_index as u64 * platform.config.execution.epoch_time_length_s * 1000
```
Comment (shumkov) [inline packages/rs-drive-abci/src/execution/platform_events/block_fee_processing/process_block_fees/v0/mod.rs:248]: Magic number, let's use constant

File: packages/rs-drive-abci/src/platform_types/epoch_info/v0/mod.rs [hunk 2]
```diff
@@ -133,13 +130,14 @@ impl EpochInfoV0 {
...
         previous_block_time_ms: Option<u64>,
+        epoch_time_s: u64,
     ) -> Result<EpochInfoV0, Error> {
         let previous_block_time = match previous_block_time_ms {
             Some(block_time) => block_time,
             None => return Ok(EpochInfoV0::default()),
         };
 
-        let epoch_change_time = Decimal::from(EPOCH_CHANGE_TIME_MS_V0);
+        let epoch_change_time = Decimal::from(epoch_time_s * 1000);
```
Comment (shumkov) [inline packages/rs-drive-abci/src/platform_types/epoch_info/v0/mod.rs:140]: Let's use a constant

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-drive-abci/src/execution/platform_events/block_fee_processing/process_block_fees/v0/mod.rs (hunk 1 above)
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/execution/platform_events/block_fee_processing/process_block_fees/v0/mod.rs:248]: What are you talking about? it's second to ms. I'm pretty sure 1000 is okay to use.

File: packages/rs-drive-abci/src/platform_types/epoch_info/v0/mod.rs (hunk 2 above)
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/platform_types/epoch_info/v0/mod.rs:140]: why? s to ms is 1000, no need for a constant.

Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/platform_types/epoch_info/v0/mod.rs:140]: (as it will never change).

Comment (shumkov) [reply packages/rs-drive-abci/src/platform_types/epoch_info/v0/mod.rs:140]: I prefer to use constants for any number especially if it’s repeating https://en.wikipedia.org/wiki/Magic_number_(programming)

It's up to you

**Review by shumkov — APPROVED** [review]

**Pull request merged**